pkg/plugin/sdk/v1alpha2/plugin/interface.go

```diff
@@ -54,3 +65,41 @@ func (*Base) PostTestHook(bool, ioutils.PromptRunner) error {
 func (*Base) PostRunHook(bool, ioutils.PromptRunner) error {
 	return nil
 }
+
+type CustomCommandFn (func(ctx context.Context, args []string) error)
+
+var fnMap map[string]CustomCommandFn = make(map[string]CustomCommandFn)
+
+type Command struct {
+	Use       string
+	ShortDesc string
+	LongDesc  string
+	Run       CustomCommandFn
+}
+
+func saveRunFunctions(commands []*Command) error {
+
+	for _, cmd := range commands {
+		if _, exists := fnMap[cmd.Use]; exists {
+			return fmt.Errorf("command '%s' is declared more than once by plugin", cmd.Use)
+		}
+		fnMap[cmd.Use] = cmd.Run
+	}
+
+	return nil
+}
+
+func executeRunFunction(command string, ctx context.Context, args []string) error {
+	return fnMap[command](ctx, args)
+}
+
+func GetWorkspaceRoot(ctx context.Context) string {
+	return ctx.Value(interceptors.WorkspaceRootKey).(string)
+}
+
+func GetBazel(ctx context.Context) bazel.Bazel {
+	workspaceRoot := GetWorkspaceRoot(ctx)
+	bzl := bazel.New()
+	bzl.SetWorkspaceRoot(workspaceRoot)
+	return bzl
+}
```

Comment:
I think these 2 functions should not exist at all. I'd argue that what we want is:

```
type CustomCommandFn (func(ctx context.Context, args []string, bzl bazel.Bazel) error)
```

And then you make it totally hidden and the user doesn't need to ask for it to be constructed.